### ISSUE TITLE
ktest --- skip does not work well as a command line option

### ISSUE THREAD

**Issue by grosu** [kframework/k-legacy#85]
If I execute ktest config.xml (my own config.xml), everything works fine.

If I execute the following, it still works fine:

fslwork$ ktest config.xml --skip pdf
Buildfile: config.xml
Kompile the language definitions...(2 in total)
Compiling 2 definitions
SUCCESS
Generating PDF documentation...(2 in total)
Skipped 2 definitions
Done.
Running /home/grosu/k/dist/tutorial/2_languages/1_simple/2_typed/1_static/exercises/exercise_2/programs/../simple-typed-static.k programs... (34 in total, 0 with input, 34 with output)
SUCCESS
Running /home/grosu/k/dist/tutorial/2_languages/1_simple/2_typed/1_static/exercises/exercise_2/programs/../simple-typed-static.k programs... (13 in total, 0 with input, 13 with output)
FAIL: ../../../../2_dynamic/exercises/exercise_2/programs/ex02.simple
FAIL: ../../../../2_dynamic/exercises/exercise_2/programs/ex03.simple
FAIL: ../../../../2_dynamic/exercises/exercise_2/programs/ex04.simple
FAIL: ../../../../2_dynamic/exercises/exercise_2/programs/ex05.simple
FAIL: ../../../../2_dynamic/exercises/exercise_2/programs/ex06.simple
FAIL: ../../../../2_dynamic/exercises/exercise_2/programs/ex07.simple
FAIL: ../../../../2_dynamic/exercises/exercise_2/programs/ex08.simple
FAIL: ../../../../2_dynamic/exercises/exercise_2/programs/ex09.simple
FAIL: ../../../../2_dynamic/exercises/exercise_2/programs/ex10.simple
FAIL: ../../../../2_dynamic/exercises/exercise_2/programs/ex11.simple
FAIL: ../../../../2_dynamic/exercises/exercise_2/programs/ex12.simple
FAIL: ../../../../2_dynamic/exercises/exercise_2/programs/ex13.simple

However, if I execute the following, then it skips krunning the examples, too:

fslwork$ ktest config.xml --skip "kompile pdf"
Buildfile: config.xml
Kompile the language definitions...(2 in total)
Skipped 2 definitions
Done.
Generating PDF documentation...(2 in total)
Skipped 2 definitions
Done.
fslwork$

**Comment by andreiarusoaie** [comment]
This happens because I did it so... If you don't kompile then I don't run the examples. This behavior is inherited from the old `ant test` where I had to make sure that definitions are compiled before running the programs. Since Jenkins  always tests a fresh copy of K where no definitions are compiled then an optimization was to run the tests only for compiled definitions. 
Now, I think it makes sense to skip that check. I'll do it when I'll arrive at school.

**Comment by grosu** [comment]
But Andrei, this did not happen a few hours ago!  I've written the README files for student HWs about 10 hours ago, while trying the commands.  See, for example:

https://github.com/kframework/k/blob/master/dist/tutorial/2_languages/1_simple/2_typed/1_static/exercises/exercise_1/programs/README

It was actually quite convenient to write --skip "kompile pdf", because it saved me a lot of time while working exclusively on the examples (and not on the definition)

**Comment by andreiarusoaie** [comment]
Oh, you are right. Apparently I allowed running the tests without compiling the definition but I completely forgot about that. The changes I did in order to allow skip in config.xml broke that. 

Now it should work. I just pushed a fix.
However, it seems that your config.xml is not correct on the repo, since it refers to an unexistent ../simple-untyped.k definition which is a few dirs up... Just to let you know. I think I've run git pull before testing.
